pkg/ccl/backupccl/restore_online.go

```diff
@@ -389,20 +389,12 @@ func (r *restoreResumer) doDownloadFiles(ctx context.Context, execCtx sql.JobExe
 	details := r.job.Details().(jobspb.RestoreDetails)
 
 	grp := ctxgroup.WithContext(ctx)
-	downloadSpansCh := make(chan roachpb.Span, len(details.DownloadSpans))
-	grp.GoCtx(func(ctx context.Context) error {
-		defer close(downloadSpansCh)
-		for _, span := range details.DownloadSpans {
-			downloadSpansCh <- span
-		}
-		return nil
-	})
+	downloadSpansCh := make(chan roachpb.Spans, 1)
 
-	restoreWorkers := int(onlineRestoreDownloadWorkers.Get(&execCtx.ExecCfg().Settings.SV))
-	for i := 0; i < restoreWorkers; i++ {
-		grp.GoCtx(r.sendDownloadWorker(execCtx, downloadSpansCh))
-	}
+	downloadSpansCh <- details.DownloadSpans
+	close(downloadSpansCh)
 
+	grp.GoCtx(r.sendDownloadWorker(execCtx, downloadSpansCh))
```

Comment:
I imagine we are keeping the context group and concurrency because waitForDownloadToComplete takes a while to calculate the total size so it is nice to get that started sooner.

But, perhaps we can simplify this a bit more by removing the channel completely now that we just have one worker?  Just thinking it is nice to keep this brand new file relatively cruft free.

Reply:
> I imagine we are keeping the context group and concurrency because waitForDownloadToComplete takes a while to calculate the total size so it is nice to get that started sooner.

I'm not sure I understand this? Both of these functions block until all the data is downloaded: 

* `waitForDownloadToComplete` polls and returns when it sees zero. It will block until the downloads are done.
* `sendDownloadWorker` only exits when the every span (or set of spans after this diff) on the ch has been sent in a download request _that got a reply_. These requests only reply when they complete the span (set of spans).

Reply:
Overlooked that we put the status update in waitForDownloadToComplete so we want that running while we wait for the workers so we get progress updates.

Looks good as is, although I guess I might still consider removing the channel.

Reply:
I was thinking I'd keep the ch since we might still decide we want to send some spans before others based on backup info or something. I donno. Can rip it out later.